content/blog/hackathon-2022-wazo-matrix-bridge.md

```diff
@@ -0,0 +1,60 @@
+---
+Title: Hackathon 2022 - Bridging Wazo and Matrix
+Date: 2022-10-20 12:00:00
+Author: Jesse Sopel
+Category: Hackathon
+Tags: Chat, Python, Matrix, Bridge, Hackathon
+Slug: hackathon-2022-bridging-wazo-and-matrix
+Status: published
+---
+
+Every year Wazo organizes a Hackathon with their employees, and as many of us as possible get together to work on some fun projects and see what we can learn.
+There were multiple teams working on different projects. Our team consisted of Charles, Francis, Francois and I (Jesse).
+We decided to both work on creating a bridge between Wazo's built-in chat system and a Matrix server, as well as [testing out adding some new features to the existing chat](https://wazo-platform.org/blog/hackaton-2022-building-group-chat-with-wazo), but this article is focused on the former.
+
+## The Objective
+
+The goal was to create a bidirectional bridge between Wazo's [Chatd](https://wazo-platform.org/documentation/overview/chat.html) and a [Matrix](https://matrix.org/) Homeserver.
+
+A user that exists on both platforms can have the two accounts mapped to each other, but in case they do not exist or have no account configured the bridge then "puppet" accounts should be created for them. This allows the Matrix user to chat with any user on Wazo as if they were using it directly. This is known as a [Double Puppeting](https://matrix.org/docs/guides/types-of-bridging#double-puppeted-bridge). Now, we didn't quite get there, but we made some good progress and learned some new things.
```

Comment:
```suggestion
A user that exists on both platforms can have the two accounts mapped to each other, but in case they do not exist or have no account configured on the bridge, then "puppet" accounts should be created for them. This allows the Matrix user to chat with any user on Wazo as if they were using it directly. This is known as a [Double Puppeting](https://matrix.org/docs/guides/types-of-bridging#double-puppeted-bridge). Now, we didn't quite get there, but we made some good progress and learned some new things.
```